### ISSUE TITLE
Tika 2.x incompatible with Quarkus Tika

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#3599]
Camel has moved on to Tika 2.x, Quarkus Tika is currently aligned with 1.x.

Seems most functionality actually works when I exclude the `org.apache.tika` dependencies from `org.apache.camel:camel-tika`, except for the `textMain` content type:

```
Caused by: java.lang.NoClassDefFoundError: org/apache/tika/sax/boilerpipe/BoilerpipeContentHandler
	at org.apache.camel.component.tika.TikaProducer.getContentHandler(TikaProducer.java:154)
	at org.apache.camel.component.tika.TikaProducer.doParse(TikaProducer.java:117)
	at org.apache.camel.component.tika.TikaProducer.process(TikaProducer.java:91)
	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:66)
	... 40 more
```

For native mode it's possible to fix and work around this via a substitution. For JVM mode it's a bit trickier to fix.

**Comment by jamesnetherton** [comment]
I have a workaround in place until things are aligned better.

**Comment by jamesnetherton** [comment]
See also https://github.com/quarkiverse/quarkus-tika/issues/22.

**Comment by zhfeng** [comment]
[TikaProducerSubstitutions.java](https://github.com/apache/camel-quarkus/blob/camel-main/extensions%2Ftika%2Fruntime%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fcamel%2Fquarkus%2Fcomponent%2Ftika%2Fgraalvm%2FTikaProducerSubstitutions.java) is missing license header.

**Comment by jamesnetherton** [comment]
> [TikaProducerSubstitutions.java](https://github.com/apache/camel-quarkus/blob/camel-main/extensions%2Ftika%2Fruntime%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fcamel%2Fquarkus%2Fcomponent%2Ftika%2Fgraalvm%2FTikaProducerSubstitutions.java) is missing license header.

I've added it now.

**Comment by zhfeng** [comment]
@jamesnetherton shoud it be closed?

**Comment by jamesnetherton** [comment]
> @jamesnetherton shoud it be closed?

I was leaving it open because the proper fix is to upgrade to a Quarkus Tika release that is aligned with Tika 2.x.

**Comment by zhfeng** [comment]
OK, I get it.

**Comment by zhfeng** [comment]
@jamesnetherton `quarkus-tika` upgrade to `2.0.0.CR1` which depends on `tika` 2.7.0 now. So I think we could revert this workaround and close this issue?

**Comment by jamesnetherton** [comment]
This was fixed in 3.0.0-M2 by https://github.com/apache/camel-quarkus/pull/4739.
